src/Modules/CalcDefence.lua

```diff
@@ -516,12 +516,19 @@ function calcs.defence(env, actor)
 	local baseBlockChance = 0
 	if actor.itemList["Weapon 2"] and actor.itemList["Weapon 2"].armourData then
 		baseBlockChance = baseBlockChance + actor.itemList["Weapon 2"].armourData.BlockChance
+		
 	end
 	if actor.itemList["Weapon 3"] and actor.itemList["Weapon 3"].armourData then
 		baseBlockChance = baseBlockChance + actor.itemList["Weapon 3"].armourData.BlockChance
-	end
+	end	
 	output.ShieldBlockChance = baseBlockChance
+	-- Override from Ascendancy or other sources
 	baseBlockChance = modDB:Override(nil, "ReplaceShieldBlock") or baseBlockChance
+	-- Bandage fix for interaction of block replacement effects with Necromantic Aegis
+	if actor == env.minion and env.player.modDB:Override(nil, "ReplaceShieldBlock") and env.allocNodes[45175] then
+		baseBlockChance = env.player.modDB:Override(nil, "ReplaceShieldBlock")
+	end
+
```

Comment:
This could just be

```suggestion
	end
	output.ShieldBlockChance = modDB:Override(nil, "ReplaceShieldBlock") or baseBlockChance

	-- Apply player block overrides if Necromantic Aegis allocated
	output.ShieldBlockChance = actor == env.minion and env.keystonesAdded["Necromantic Aegis"] and env.player.modDB:Override(nil, "ReplaceShieldBlock") or output.ShieldBlockChance

```

using keystonesAdded will be more flexible if the keystone id changes.

Reply:
Ah nice catch, Ill keep that in mind next time a similar issue comes up. Ill fix it soon.

Reply:
You should have an apply button somewhere around if i did this right.